packages/xstate-wallet/src/workflows/create-and-fund-ledger.ts

```diff
@@ -0,0 +1,184 @@
+};
+
+const initializeChannel = (
+  store: Store
+): WorkflowServices['initializeChannel'] => async context => {
+  const stateVars: StateVariables = {
+    outcome: context.initialOutcome,
+    turnNum: bigNumberify(0),
+    isFinal: false,
+    appData: '0x0'
+  };
+  const entry = await store.createChannel(context.participants, CHALLENGE_DURATION, stateVars);
+  return entry.channelId;
+};
+
+const createObjective = (store: Store): WorkflowServices['createObjective'] => async context => {
+  const objective: Objective = {
+    type: 'FundLedger',
+    participants: context.participants,
+    data: {ledgerId: context.ledgerId}
+  };
+  return store.addObjective(objective);
+};
+const getPreFundState = (store: Store): WorkflowServices['getPreFundState'] => async context => {
+  const {latestState} = await store.getEntry(context.ledgerId);
+  return {state: latestState};
+};
+// TODO: Can this be shared with create-and-fund?
+const getDepositingInfo = (
+  store: Store
+): WorkflowServices['getDepositingInfo'] => async context => {
+  const {supported, myIndex} = await store.getEntry(context.ledgerId);
+  const {allocationItems} = checkThat(supported?.outcome, isSimpleEthAllocation);
+
+  const fundedAt = allocationItems.map(a => a.amount).reduce(add);
+  let depositAt = bigNumberify(0);
+  for (let i = 0; i < allocationItems.length; i++) {
+    const {amount} = allocationItems[i];
+    if (i !== myIndex) depositAt = depositAt.add(amount);
+    else {
+      const totalAfterDeposit = depositAt.add(amount);
+      return {channelId: context.ledgerId, depositAt, totalAfterDeposit, fundedAt};
+    }
+  }
+
+  throw Error(`Could not find an allocation for participant id ${myIndex}`);
+};
+
+export const options = (
+  store: Store
+): {actions: WorkflowActions; services: WorkflowServices; guards: WorkflowGuards} => {
+  return {
+    actions: {
+      assignChannelId: assign((context: WorkflowContext, event: DoneInvokeEvent<string>) => {
+        return {...context, channelId: event.data};
+      })
+    },
+    guards: {
+      doesChannelIdExist: (context: WorkflowContext) => {
+        return !!context.ledgerId;
+      }
+    },
+    services: {
+      initializeChannel: initializeChannel(store),
+      supportState: SupportState.machine(store),
+      createObjective: createObjective(store),
+      depositing: Depositing.machine(store),
+      getDepositingInfo: getDepositingInfo(store),
+      getPreFundState: getPreFundState(store)
+    }
+  };
+};
+
+export const mockGuards: WorkflowGuards = {
+  doesChannelIdExist: () => true
+};
+export const mockActions: WorkflowActions = {
+  assignChannelId: 'assignChannelId' as any
+};
+export const mockServices: WorkflowServices = {
+  initializeChannel: 'initializeChannel' as any,
+  supportState: 'supportState' as any,
+  createObjective: 'createObjective' as any,
+  depositing: 'depositing' as any,
+  getDepositingInfo: 'getDepositingInfo' as any,
+  getPreFundState: 'getPreFundState' as any
+};
```

Comment:
Are these needed? (and do they need to be exported?)
```suggestion
```

Reply:
I thought they were required by the `generateConfig` to create the config for the visualizer?

Reply:
Only guards; if you try to execute with missing `actions` or `services`, then the interpreter just logs warnings